src/AbstractDifferentiation.jl

```diff
@@ -224,63 +247,200 @@ struct LazyDerivative{B, F, X}
     f::F
     xs::X
 end
+
 function Base.:*(d::LazyDerivative, y)
-    return derivative(d.ab, d.f, d.xs...) * y
+    return derivative(d.backend, d.f, d.xs...) * y
 end
+
 function Base.:*(y, d::LazyDerivative)
-    return y * derivative(d.ab, d.f, d.xs...)
+    return y * derivative(d.backend, d.f, d.xs...)
+end
+
+function Base.:*(d::LazyDerivative, y::Union{Number,Tuple})
```

Comment:
I don't think we should support multiplication if the derivative returns a tuple. This part of the code can be simplified.

Reply:
Don't we want to support functions like `fder(x, y) = exp(y) * x + y * log(x)` for the LazyOperators as well?

Reply:
Hmm I guess we could.